pkgs/dart_tooling_mcp_server/test/tools/dtd_test.dart

```diff
@@ -75,4 +87,55 @@ void main() {
       );
     });
   });
+
+  group('$VmService management', () {
+    setUp(() async {
+      testHarness = await TestHarness.start(inProcess: true);
+      await testHarness.connectToDtd();
+    });
+
+    test('persists vm services', () async {
+      final server = testHarness.serverConnectionPair.server!;
+      expect(server.activeVmServices, isEmpty);
+
+      await testHarness.startDebugSession(
+        counterAppPath,
+        'lib/main.dart',
+        isFlutter: true,
+      );
+      await server.updateActiveVmServices();
+      expect(server.activeVmServices.length, 1);
+
+      // Re-uses existing VM Service when available.
+      final originalVmService = server.activeVmServices.values.single;
+      await server.updateActiveVmServices();
+      expect(server.activeVmServices.length, 1);
+      expect(originalVmService, server.activeVmServices.values.single);
+
+      // await testHarness.startDebugSession(
+      //   counterAppPath,
+      //   'lib/main.dart',
+      //   isFlutter: true,
+      // );
+      // await server.updateActiveVmServices();
+      // expect(server.activeVmServices.length, 2);
```

Comment:
@jakemac53 when I uncomment this code, I get the following error:

```
  Bad state: Unable to find the context to /Users/kenzieschmoll/develop/dart-lang/ai/pkgs/dart_tooling_mcp_server/test_fixtures/counter_app/.dart_tool/dartpad/web_plugin_registrant.dart
  package:analyzer/src/dart/analysis/analysis_context_collection.dart 145:5  AnalysisContextCollectionImpl.contextFor
  package:dart_tooling_mcp_server/src/mixins/analyzer.dart 98:17             DartAnalyzerSupport._createAnalysisContext.<fn>
```

Is this expected or is this a bug in the test harness? With the current implementation should we expect running a second debug session to fail?

Reply:
It sounds like possibly an error somewhere... `startDebugSession` does add a new root dir to the client - and the server will then ask for the new roots and set up new file listeners.

I think this might be hitting a race condition where we are getting a file change notification after the context has been disposed and before the new one for that dir has been set up.

Reply:
Hmm, looking into this a bit and I wonder if its actually that the path is not analyzed by default because its in a hidden dir?

Reply:
For now maybe its best to just guard this in a try/catch https://github.com/dart-lang/ai/blob/main/pkgs/dart_tooling_mcp_server/lib/src/mixins/analyzer.dart#L97

Reply:
Added a try / catch and filed https://github.com/dart-lang/ai/issues/65.